qt/python/mantidqt/dialogs/spectraselectorutils.py

```diff
@@ -24,9 +24,17 @@ def get_spectra_selection(workspaces, parent_widget=None, show_colorfill_btn=Fal
     """
     SpectraSelectionDialog.raise_error_if_workspaces_not_compatible(workspaces)
     single_spectra_ws = [wksp.getNumberHistograms() for wksp in workspaces if wksp.getNumberHistograms() == 1]
-    if len(single_spectra_ws) > 0 and len(workspaces) == 1:
-        # At least 1 workspace contains only a single spectrum so this is all
-        # that is possible to plot for all of them
+
+    ws_spectra = [{ws.getSpectrum(i).getSpectrumNo() for i in range(ws.getNumberHistograms())} for ws in workspaces]
```

Comment:
I think in the case where all workspaces have a single spectrum then it is correct to simply overplot them all. Could we avoid this potentially expensive loop in that scenario?